sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/reflection/KalixMethod.scala

```diff
@@ -95,12 +95,15 @@ case class VirtualServiceMethod(component: Class[_], methodName: String, inputTy
  * Build from methods annotated with @Subscription. Those methods are not annotated with Spring REST annotations, but
  * they become a REST method at the end.
  */
-case class RestServiceMethod(javaMethod: Method, inputTypeParamIndex: Int = 0) extends AnyServiceMethod {
+case class RestServiceMethod(
+    javaMethod: Method,
+    inputTypeParamIndex: Int = 0,
+    methodName: String,
+    methodsMap: Map[String, Method] = Map())
+    extends AnyServiceMethod {
```

Comment:
I don't know why we called it a `RestServiceMethod`. We should have call it a `SubscriptionServiceMethod`. 

And because they will serve a very specific purpose, we can eventually remove the `javaMethod` and only have a `methodsMap`. 

We can then have a map with only one entry and always use the typeUrl to look it up. 

There is one potential risk though. When the user defines one single method and mess up with the typeUrl, we may not even find that single method when doing the look up. 

As a workaround, we could have a best-effort strategy. If we don't find a match, but we know that we only have one entry in the map, we try to deliver the payload to that single method.

Reply:
I agree with your approach I wanted to leave these changes for the refactoring we talked about. I started to do precisely what you are suggesting and started to get a bit messy so I decided to leave until then.

Reply:
done